components/motor/gpio/motor_encoder.go

```diff
@@ -354,184 +240,162 @@ func (m *EncodedMotor) rpmMonitor() {
 			continue
 		}
 		now := time.Now().UnixNano()
-		if now == lastTime {
-			// this really only happens in testing, b/c we decrease sleep, but nice defense anyway
+		atomic.AddInt64(&m.rpmMonitorCalls, 1)
+
+		if (m.DirectionMoving() == 1 && pos >= m.state.goalPos) || (m.DirectionMoving() == -1 && pos <= m.state.goalPos) {
+			// stop motor when at or past goal position
+			if err := m.Stop(m.cancelCtx, nil); err != nil {
+				m.logger.Error(err)
+				return
+			}
 			continue
 		}
-		atomic.AddInt64(&m.rpmMonitorCalls, 1)
-		// TODO: we round down here for absolute encoders, but absolute encoders
-		// should have their own logic separate from incremental
-		roundedPos := int64(math.Floor(pos))
-		inRamp = m.rpmMonitorPass(roundedPos, lastPos, now, lastTime, rpmDebug)
 
-		lastPos = int64(pos)
+		m.makeAdjustments(pos, lastPos, now, lastTime)
+
+		lastPos = pos
 		lastTime = now
 	}
 }
 
-// return is if we are in a ramp phase.
-func (m *EncodedMotor) rpmMonitorPass(pos, lastPos, now, lastTime int64, rpmDebug bool) bool {
+// makeAdjustments does the math required to see if the RPM is too high or too low,
+// and if the goal position has been reached.
+func (m *EncodedMotor) makeAdjustments(pos, lastPos float64, now, lastTime int64) {
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
 
-	var ticksLeft int64
-
-	currentRPM := m.computeRPM(pos, lastPos, now, lastTime)
-	m.state.currentRPM = currentRPM
-
-	if !m.state.regulated && math.Abs(m.state.desiredRPM) > 0.001 {
-		m.rpmMonitorPassSetRpmInLock(currentRPM, m.state.desiredRPM, -1, rpmDebug)
-		return false
+	// calculate RPM based on change in position and change in time
+	deltaPos := (pos - lastPos) / float64(m.ticksPerRotation)
+	deltaTime := (float64(now) - float64(lastTime)) / float64(6e10)
```

Comment:
what's the 6e10 value? Is the delta time calculated in minutes?

Reply:
converting from nanoseconds to minutes. I'm assuming it's originally in nanoseconds since there is so little time between calls?

Reply:
let's see if we start with polling the time minutes and don't covert? then `currentRPM` will almost always be 0 causing us to keep increasing the power when we don't need to

Reply:
no - sorry - I'm reading this line as : we are polling in minutes, if we are polling in nanoseconds, then I'm not concerned.

Reply:
let's add a comment to let us know that it is converting minutes to nano seconds in this line, and we are polling at that time magnitude.